python/dgl/distributed/rpc.py

```diff
@@ -338,6 +550,73 @@ def send_request(target, request):
     msg = RPCMessage(service_id, msg_seq, client_id, server_id, data, tensors)
     send_rpc_message(msg)
 
+def recv_request(timeout=0):
+    """Receive one request.
+
+    Receive one :class:`RPCMessage` and de-serialize it into a proper Request object.
+
+    The operation is blocking -- it returns when it receives any message
+    or it times out.
+
+    Parameters
+    ----------
+    timeout : int, optional
+        The timeout value in milliseconds. If zero, wait indefinitely.
+
+    Returns
+    -------
+    req : request
+        One request received from the target, or None if it times out.
+
+    Raises
+    ------
+    ConnectionError if there is any problem with the connection.
+    """
+    # TODO(chao): handle timeout
+    msg = recv_rpc_message(timeout)
+    if msg is None:
+        return None
+    req_cls, _ = SERVICE_ID_TO_PROPERTY[msg.service_id]
```

Comment:
why is this different from the one in `recv_response`?

Reply:
req_cls and res_cls are different classes.

Reply:
from the code perspective, the difference is just the name?

Reply:
Here are some logic difference. For example, the recv_response will not invoke set_client_id, set_server_id, and set_msg_seq functions.